### ISSUE TITLE
Don't install interstitial page on inactive pinned tab

### ISSUE THREAD

**Issue by sangwoo108** [brave/brave-browser#38510]
Reported from https://github.com/brave/brave-browser/issues/38444#issuecomment-2125985309 by @TEMP-ad

This is causing any tab pinned or unpinned to get the interstitial message.

1.68.36 Chromium: 125.0.6422.76 (Official Build) nightly (64-bit)  Windows 11 Version 23H2 (Build 22631.3593)

How to reproduce:

1. just have some tabs opened
2. pin some
3.  open a new window
4. click on the pinned tabs in the new window.
5. see how the active tab in the original window gets the message about the tab being shared
6. the only way is to go to the original window and change tabs, if you click on the interstitial message nothing will happen, it will only switch between pinned tabs if you are on the correct pinned tab.
 
https://github.com/brave/brave-browser/assets/156715894/9b54cd74-3bbc-48bb-a4b3-f24ebce51c85

**Comment by TEMP-ad** [comment]
It works better! nice fix, but there is still what I think a behavior that shouldn't happen, and this will be more frequent with multiple monitors setups.

https://github.com/brave/brave-browser/assets/156715894/49122328-d8ce-4000-98d4-d4c8a46e4afc

As you can see, if you have both pinned tabs active in both windows, and then you click on random tab (pinned or unpinned) in the other window that has the interstitial page / not the active one, the interstitial page will still be installed in that window, even if you never clicked that specific pinned tab to make it active or anything.

I don't know the technicity about this and if this can be fixed, because I know just moving the window or clicking on a button or menu or anything, will make the tab active 
example:

https://github.com/brave/brave-browser/assets/156715894/f1861077-2ba9-43cf-93e2-a88eea10ad57

Of course, clicking on a background tab in the right window, to make it active but still want the pinned tab on the left to stay active, like if you have a video or something, would be a real scenario, where you wouldn't want the pinned tab to receive the active status and the one that is still visible get the interstitial page.

**Comment by sangwoo108** [comment]
Thanks @TEMP-ad for detailed explanation, as always. 

> As you can see, if you have both pinned tabs active in both windows, and then you click on random tab (pinned or unpinned) in the other window that has the interstitial page / not the active one, the interstitial page will still be installed in that window, even if you never clicked that specific pinned tab to make it active or anything.

I guess that's because the pinned tab is moved to the newly activated window first, and then another tab is activated. Unfortunately, I think it'd be hard to fix as we can't know if the window activation will cause the active tab to be changed ahead of time.

**Comment by TEMP-ad** [comment]
@sangwoo108 thanks for the answer! I was afraid about it the moment I saw that clicking on anything would bring the content.

But if that's the case (and maybe I am just seeing this with a magnifier, since I judged the behavior based on the text) shouldn't the description matches the behavior? so instead of tab it should say window?
Like: 

> Clicking this **tab** will bring its content to this window

vs

> Clicking this **window** will bring its content

Or whatever sounds good but has the word **window** instead of _tab_? that would be the way avoid confusion since it is exactly what it does, where clicking anywhere in the window will just bring its content.

Since it is complicated to fix, then fixing the text seems to be the easiest thing in my mind, then.
Also, just hitting alt+tab will work when this happens anyway, so no real issues. guess I just read too much the _instructions_ as well. 😁

Either way, thanks for the fix! love the feature. 👍
